src/autotester/server/utils/tests/test_file_management.py

```diff
@@ -0,0 +1,176 @@
+
+    def test_student_files(self, t_path, f_path, args):
+        """
+        Checks whether the permission of files and directories
+        in student files changed into '0o770' and '0o660'
+        """
+        markus_address, assignment_id = args
+        tests_path = t_path
+        files_path = f_path
+        test_username = Path(tests_path).owner()
+        student_files, script_files = setup_files(
+            files_path, tests_path, test_username, markus_address, assignment_id
+        )
+        stdir_pmsn = []
+        stfile_pmsn = []
+        for fd, file_or_dir in student_files:
+            if fd == "d":
+                permission = fd_permission(file_or_dir)
+                stdir_pmsn.append(
+                    True
+                ) if permission == "-rwxrwx---" else stdir_pmsn.append(False)
+            else:
+                permission = fd_permission(file_or_dir)
+                stfile_pmsn.append(
+                    True
+                ) if permission == "-rw-rw----" else stfile_pmsn.append(False)
+        assert all(i for i in stdir_pmsn)
+        assert all(i for i in stfile_pmsn)
+
+    def test_script_files(self, t_path, f_path, args):
+        """
+        Checks whether the permission of files and directories
+        in script files changed into '0o1770' and '0o640'
+        """
+        markus_address, assignment_id = args
+        tests_path = t_path
+        files_path = f_path
+        test_username = Path(tests_path).owner()
+        student_files, script_files = setup_files(
+            files_path, tests_path, test_username, markus_address, assignment_id
+        )
+        scdir_pmsn = []
+        scfile_pmsn = []
+        for fd, file_or_dir in script_files:
+            if fd == "d":
+                permission = fd_permission(file_or_dir)
+                scdir_pmsn.append(
+                    True
+                ) if permission == "drwxrwx--T" else scdir_pmsn.append(False)
+            else:
+                mode = os.stat(file_or_dir).st_mode
+                permission = stat.filemode(mode)
+                scfile_pmsn.append(
+                    True
+                ) if permission == "-rw-r-----" else scfile_pmsn.append(False)
+        assert all(i for i in scdir_pmsn)
+        assert all(i for i in scfile_pmsn)
+
+    def test_copied_file(self, t_path, f_path, args):
+        """
+        Checks whether all the copied files are exists
+        """
+        markus_address, assignment_id = args
+        tests_path = t_path
+        files_path = f_path
+        test_username = Path(tests_path).owner()
+        student_files, script_files = setup_files(
+            files_path, tests_path, test_username, markus_address, assignment_id
+        )
+        st_files = []
+        scr_files = []
+        [st_files.append(i[1]) for i in student_files]
+        [scr_files.append(j[1]) for j in script_files]
+        assert all(os.path.exists(f) for f in st_files)
```

Comment:
As above, use an explicit loop and `assert` statement inside the loop body.

Also, it's better here to use the original `tests_path` and `files_path` to determine what files to actually check the existence of.